lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -0,0 +1,326 @@
+                    None,
+                    "B",
+                    [
+                        self.make_expected_tree("courseware-3", "1"),
+                        self.make_expected_tree("courseware-4", "2"),
+                    ]
+                ),
+                self.make_expected_tree(
+                    None,
+                    "C",
+                    [self.make_expected_tree("courseware-5", "1")]
+                ),
+            ],
+            "non_courseware_topics": [
+                self.make_expected_tree("non-courseware-1", "A"),
+                self.make_expected_tree("non-courseware-2", "B"),
+            ],
+        }
+        self.assertEqual(actual, expected)
+
+    def test_sort_key(self):
+        self.make_discussion_module("courseware-1", "First", "A", sort_key="D")
+        self.make_discussion_module("courseware-2", "First", "B", sort_key="B")
+        self.make_discussion_module("courseware-3", "First", "C", sort_key="E")
+        self.make_discussion_module("courseware-4", "Second", "A", sort_key="F")
+        self.make_discussion_module("courseware-5", "Second", "B", sort_key="G")
+        self.make_discussion_module("courseware-6", "Second", "C")
+        self.make_discussion_module("courseware-7", "Second", "D", sort_key="A")
+        self.course.discussion_topics = {
+            "W": {"id": "non-courseware-1", "sort_key": "Z"},
+            "X": {"id": "non-courseware-2"},
+            "Y": {"id": "non-courseware-3", "sort_key": "Y"},
+            "Z": {"id": "non-courseware-4", "sort_key": "W"},
+        }
+        self.course.save()
+        actual = self.get_course_topics()
+        expected = {
+            "courseware_topics": [
+                self.make_expected_tree(
+                    None,
+                    "First",
+                    [
+                        self.make_expected_tree("courseware-2", "B"),
+                        self.make_expected_tree("courseware-1", "A"),
+                        self.make_expected_tree("courseware-3", "C"),
+                    ]
+                ),
+                self.make_expected_tree(
+                    None,
+                    "Second",
+                    [
+                        self.make_expected_tree("courseware-7", "D"),
+                        self.make_expected_tree("courseware-6", "C"),
+                        self.make_expected_tree("courseware-4", "A"),
+                        self.make_expected_tree("courseware-5", "B"),
+                    ]
+                ),
+            ],
+            "non_courseware_topics": [
+                self.make_expected_tree("non-courseware-4", "Z"),
+                self.make_expected_tree("non-courseware-2", "X"),
+                self.make_expected_tree("non-courseware-3", "Y"),
+                self.make_expected_tree("non-courseware-1", "W"),
+            ],
+        }
+        self.assertEqual(actual, expected)
+
+    def test_access_control(self):
+        """
```

Comment:
The different endpoints will do different things based on the user's level of access. I think it makes the most sense to factor out common code as the later endpoints are written instead of speculating.